core/tests/org.openjdk.jmc.jdp.test/src/test/java/org/openjdk/jmc/jdp/client/TestToolkit.java

```diff
@@ -148,4 +156,37 @@ public int getMaxHeartBeatTimeout() {
 		}
 		return bytes;
 	}
+
+    private static boolean isBroadcastingSupported() {
+        try {
+            InetAddress multiCastAddress = InetAddress.getByName("239.255.255.255");
+            int multiCastPort = 7711;
+            Thread thread = new Thread(() -> {
+                try {
+                    MulticastSocket ssocket = new MulticastSocket(multiCastPort);
+                    ssocket.setTimeToLive(1);
+                    ssocket.joinGroup(multiCastAddress);
+                    final DatagramPacket dp = new DatagramPacket(new byte[]{1}, 1, multiCastAddress, multiCastPort);
+                    while (true) {
+                        ssocket.send(dp);
+                        Thread.sleep(10);
+                    }
+                } catch (InterruptedException | IOException e) {
+                }
+            });
+            thread.start();
+            try {
```

Comment:
Same here, try-with resources. The thread.interrupt() can then be done in the `finally` clause.